src/plugin/xyz/xyzproviderimport.js

```diff
@@ -0,0 +1,206 @@
+  }
+
+  /**
+   * @inheritDoc
+   */
+  accept() {
+    if (!this.scope['form']['$invalid']) {
+      this.cleanConfig();
+
+      this.dp = this.getDataProvider();
+      this.dp.setEditable(true);
+
+      this.scope['config']['description'] = 'URL: ' + this.scope['config']['url'] +
+        '<br>Projection: ' + this.scope['config']['projection']['code'] +
+        '<br>Minimum Zoom: ' + this.scope['config']['minZoom'] +
+        '<br>Maximum Zoom: ' + this.scope['config']['maxZoom'] +
+        '<br>Zoom Offset: ' + this.scope['config']['zoomOffset'];
+      this.createXYZDescriptor(this.scope['config']);
+      this.close();
+    }
+  }
+
+  /**
+   * @inheritDoc
+   */
+  getDataProvider() {
+    const dp = ConfigProvider.create('xyz', {'label': 'XYZ Layers', 'listInServers': false});
+    return dp;
+  }
+
+  /**
+   * Create an XYZ descriptor from a layer config.
+   * @param {!Object} layerConfig The layer config.
+   * @return {IDataDescriptor} The descriptor, or null if one could not be created.
+   */
+  createXYZDescriptor(layerConfig) {
+    const projectionObject = layerConfig['projectionObject'] ? layerConfig['projectionObject'] :
+      layerConfig['projection'];
+    if (projectionObject) {
+      const projectionCode = projectionObject['code'];
+      layerConfig['projectionObject'] = projectionObject;
+      layerConfig['projection'] = projectionCode;
+      layerConfig['title'] = layerConfig['label'];
+    }
+
+    let descriptor = null;
+    descriptor = new ConfigDescriptor();
+    descriptor.setBaseConfig(layerConfig);
+    osLayer.createFromOptions(layerConfig);
+
+    this.saveXYZDescriptor(descriptor);
+
+    return descriptor;
+  }
+
+  /**
+   * Save an XYZ descriptor to the XYZ provider.
+   * @param {!IDataDescriptor} descriptor The descriptor.
+   */
+  saveXYZDescriptor(descriptor) {
+    if (descriptor) {
+      const layerConfig = descriptor.getBaseConfig();
+
+      if (this.dp) {
+        this.dp.setEditable(true);
+
+        // Set config descriptor options.
+        const id = `${getRandomString()}`;
+        layerConfig['nodeUi'] = '<descriptornodeui></descriptornodeui>';
+        layerConfig['provider'] = this.dp.getLabel();
+        layerConfig['id'] = `xyz${ID_DELIMITER}${id}`;
+
+        // Prevent the data manager from automatically expiring the descriptor.
+        descriptor.setLocal(true);
+
+        this.dp.addLayerGroup(id, layerConfig);
+      }
+    }
+  }
+
+  /**
+   * @return {string}
+   */
+  getLabel() {
+    if (this.dp) {
+      var label = /** @type {ConfigProvider} */ (this.dp).getLabel();
+      return label || '';
+    }
+
+    return '';
+  }
+
+  /**
+   * Handle projection change
+   * @export
+   */
+  onProjectionChange() {
+    this.scope['config']['zoomOffset'] = null;
+    switch (this.scope['config']['projection'].code) {
+      case 'EPSG:3857':
+        this.scope['config']['zoomOffset'] = 0;
```

Comment:
We might want to make the 3857 case also fall through as the default. We don't configure other projections by default anyway, but I think 0 zoom offset is a good default behavior.